### PR TITLE
Add default balance for not found addresses when checking collateral

### PR DESCRIPTION
**Pull request by kriscieslak**

### CODE REVIEW

**Review by marcinwadon — CHANGES_REQUESTED** [review]
I think it should be labelled as `fix`, not `feature`

File: modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/collateral/Collateral.scala [hunk 1]
```diff
@@ -23,11 +26,13 @@ object Collateral {
...
         for {
           address <- peerId.value.toPublicKey.map(_.toAddress)
-          balances <- latestBalances.getLatestBalances
-        } yield
-          balances
-            .flatMap(_.get(address))
-            .map(_.value >= config.amount.value)
-            .getOrElse(false)
+          balance <- getBalance(address)
+        } yield balance.value >= config.amount.value
```
[the remark this answers is not in the document]
Comment (kriscieslak) [reply modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/collateral/Collateral.scala]: LGTM. Changed.

File: modules/sdk/src/test/scala/org/tessellation/sdk/infrastructure/collateral/CollateralSuite.scala [hunk 2]
```diff
@@ -32,34 +32,50 @@ object CollateralSuite extends MutableIOSuite with Checkers {
...
-    mkCollateral(Some(Map((address1, Balance(NonNegLong(25_000_000_000_000L))))))
+    mkCollateral(Some(Map((address1, Balance(NonNegLong(25_000_000_000_000L))))), Amount(25_000_000_000_000L))
       .hasCollateral(peer1)
       .map(expect.same(true, _))
   }
 
   test("should return false when the balance for a given address is lower than the required collateral") {
     implicit sc =>
-      mkCollateral(Some(Map((address1, Balance(NonNegLong(24_999_999_999_999L))))))
+      mkCollateral(Some(Map((address1, Balance(NonNegLong(24_999_999_999_999L))))), Amount(25_000_000_000_000L))
```
Comment (tbekas) [inline modules/sdk/src/test/scala/org/tessellation/sdk/infrastructure/collateral/CollateralSuite.scala:78]: Same here

File: modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/collateral/Collateral.scala [hunk 3]
```diff
@@ -21,13 +23,15 @@ object Collateral {
...
-          balances <- latestBalances.getLatestBalances
-        } yield
-          balances
-            .flatMap(_.get(address))
-            .map(_.value >= config.amount.value)
-            .getOrElse(false)
+        peerId.value.toPublicKey
+          .map(_.toAddress)
+          .flatMap(getBalance)
+          .map(_.value >= config.amount.value)
```
[the remark this answers is not in the document]
Comment (kriscieslak) [reply modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/collateral/Collateral.scala]: Added just `order` for `Amount` as there is implicit conversion from `Balance` to `Amount`. Also had to add return type to `getBalance` method to use this conversion.

File: modules/sdk/src/test/scala/org/tessellation/sdk/infrastructure/collateral/CollateralSuite.scala [hunk 4]
```diff
@@ -32,34 +32,50 @@ object CollateralSuite extends MutableIOSuite with Checkers {
...
+    "should return true when the balance for a given address is empty or not found and required collateral is 0"
+  ) { implicit sc =>
+    mkCollateral(Some(Map.empty), Amount(0L))
+      .hasCollateral(peer1)
+      .map(expect.same(true, _))
+  }
+
   test("should return true when the balance for a given address is equal to the required collateral") { implicit sc =>
-    mkCollateral(Some(Map((address1, Balance(NonNegLong(25_000_000_000_000L))))))
+    mkCollateral(Some(Map((address1, Balance(NonNegLong(25_000_000_000_000L))))), Amount(25_000_000_000_000L))
```
[the remark this answers is not in the document]
Comment (kriscieslak) [reply modules/sdk/src/test/scala/org/tessellation/sdk/infrastructure/collateral/CollateralSuite.scala]: Fixed.

File: modules/sdk/src/test/scala/org/tessellation/sdk/infrastructure/collateral/CollateralSuite.scala (hunk 2 above)
Comment (kriscieslak) [reply modules/sdk/src/test/scala/org/tessellation/sdk/infrastructure/collateral/CollateralSuite.scala]: Fixed.

**Review by tbekas — APPROVED** [review]
LGTM

**Pull request merged**